Content.IntegrationTests/Tests/Administration/Logs/AdminLogTests.cs

```diff
@@ -0,0 +1,257 @@
+        });
+
+        await WaitUntil(server, async () =>
+        {
+            var logs = sAdminLogSystem.AllJson(new LogFilter
+            {
+                Round = sGameTicker.RoundId,
+                Search = guid.ToString()
+            });
+
+            await foreach (var json in logs)
+            {
+                var root = json.RootElement;
+
+                Assert.That(root.TryGetProperty("Entity", out _), Is.True);
+
+                json.Dispose();
+
+                return true;
+            }
+
+            return false;
+        });
+    }
+
+    [Test]
+    public async Task AddAndGetUnformattedLog()
+    {
+        var server = StartServer(new ServerContentIntegrationOption
+        {
+            CVarOverrides =
+            {
+                [CCVars.AdminLogsQueueSendDelay.Name] = "0"
+            }
+        });
+        await server.WaitIdleAsync();
+
+        var sDatabase = server.ResolveDependency<IServerDbManager>();
+        var sEntities = server.ResolveDependency<IEntityManager>();
+        var sMaps = server.ResolveDependency<IMapManager>();
+        var sSystems = server.ResolveDependency<IEntitySystemManager>();
+
+        var sAdminLogSystem = sSystems.GetEntitySystem<AdminLogSystem>();
+        var sGameTicker = sSystems.GetEntitySystem<GameTicker>();
+
+        var guid = Guid.NewGuid();
+
+        await server.WaitPost(() =>
+        {
+            var coordinates = GetMainEntityCoordinates(sMaps);
+            var entity = sEntities.SpawnEntity(null, coordinates);
+
+            sAdminLogSystem.Add(LogType.Unknown, $"{entity} test log: {guid}");
+        });
+
+        LogRecord log = null;
+
+        await WaitUntil(server, async () =>
+        {
+            var logs = sAdminLogSystem.All(new LogFilter
+            {
+                Round = sGameTicker.RoundId,
+                Search = guid.ToString()
+            });
+
+            await foreach (var found in logs)
+            {
+                log = found;
+                return true;
+            }
+
+            return false;
+        });
+
+        await server.WaitPost(() =>
+        {
+            Task.Run(async () =>
+            {
+                var filter = new LogFilter
+                {
+                    Round = log.RoundId,
+                    Search = log.Message,
+                    Types = new List<LogType> {log.Type}
+                };
+
+                await foreach (var json in sDatabase.GetAdminLogsJson(filter))
+                {
+                    var root = json.RootElement;
+
+                    Assert.That(root.TryGetProperty("entity", out _), Is.True);
+                    Assert.That(root.TryGetProperty("guid", out _), Is.True);
+
+                    json.Dispose();
+                }
+            }).Wait();
+        });
+    }
+
+    [Test]
+    [TestCase(500, false)]
+    [TestCase(500, true)]
+    public async Task AddLogs(int amount, bool parallel)
```

Comment:
feels like this should be a benchmark and not a test?

Reply:
This is mostly to test that bulk-adding logs (specially in parallel) doesn't cause in any errors, but I can do that too.